exasol_transformers_extension/udfs/models/filling_mask_udf.py

```diff
@@ -108,16 +110,19 @@ def load_models(self, model_name: str, **kwargs) -> None:
 
     def setup_pipeline(self, **kwargs) -> None:
         """
-        Setup pipeline with the loaded models and the current topk value
-        """
-
-        self.last_created_pipeline = self.pipeline(
-            "fill-mask",
-            model=self.last_loaded_model,
-            tokenizer=self.last_loaded_tokenizer,
-            device=self.device,
-            framework="pt",
-            top_k=kwargs['top_k'])
+        Setup pipeline if new models are loaded or if the top_k value
+        for the same model changes
+        """
+        top_k = kwargs['top_k']
+        if self.last_used_top_k is None or self.last_used_top_k != top_k:
+            self.last_created_pipeline = self.pipeline(
```

Comment:
to test this, we probably would need to count the number of calls to pipeline() in the Mock, but I think, we had already a ticket for that

Reply:
yes, we have https://github.com/exasol/transformers-extension/issues/24